flink-table/flink-table-runtime-blink/src/main/java/org/apache/flink/table/runtime/aggregate/over/frame/OverWindowFrame.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.table.runtime.aggregate.over.frame;
+
+import org.apache.flink.table.dataformat.BaseRow;
+import org.apache.flink.table.dataformat.BinaryRow;
+import org.apache.flink.table.runtime.context.ExecutionContext;
+import org.apache.flink.table.runtime.util.ResettableExternalBuffer;
+
+import java.io.Serializable;
+
+/**
+ * A window frame calculates the results for those records belong to a window frame.
+ * Before use a frame must be prepared by passing it all the records in the current partition.
+ */
+public interface OverWindowFrame extends Serializable {
+
+	void open(ExecutionContext ctx) throws Exception;
+
+	void resetBuffer(ResettableExternalBuffer rows) throws Exception;
+
+	//return the ACC of the window frame.
+	BaseRow write(int index, BaseRow current) throws Exception;
```

Comment:
what does `current` row used for? and what is `curren` row?

Reply:
Method `write` need modify to `process`.

Over AGG means that every Row has a corresponding output.
OverWindowFrame is called by:
1. Get all data and invoke `prepare(Resettable External Buffer rows)` for partition
2. Then each Row is traversed one by one to invoke `BaseRow process(int index, BaseRow current)`to get the calculation results of the currentRow.